src/Symfony/Component/Validator/Constraints/UrlValidator.php

```diff
@@ -64,6 +64,15 @@ public function validate(mixed $value, Constraint $constraint): void
             return;
         }
 
+        if ($constraint->requireTld && !str_contains($value, '.')) {
```

Comment:
Did that not open the door to the following "false-positive" ? 

- http://foo/bar.png
- http://user.pass@local
- http://127.0.0.1
- '../foo'

Reply:
Good catch! I've updated the validation logic. Thanks for the review!